### PR TITLE
feat: minecraft 1.21.4

### PR DESCRIPTION
**Pull request by 0utplay**
### Motivation
MC 1.21.4 was released a few days ago. We want to update our platforms to support it properly.

### Modification
Updated downloads, packet events (for npcs) and minestom

### Result
1.21.4 is supported on Fabric, Bukkit, Velocity, Bungee and 1.21.3 on minestom

### CODE REVIEW

File: modules/bridge/src/main/java/eu/cloudnetservice/modules/bridge/platform/minestom/MinestomBridgeManagement.java [hunk 1]
```diff
@@ -126,12 +134,15 @@ public void registerServices(@NonNull ServiceRegistry registry) {
...
+    return this.permissionChecker.get()::hasPermission;
   }
 
   @Override
   public boolean isOnAnyFallbackInstance(@NonNull Player player) {
-    return this.isOnAnyFallbackInstance(this.ownNetworkServiceInfo.serverName(), null, player::hasPermission);
+    return this.isOnAnyFallbackInstance(
+      this.ownNetworkServiceInfo.serverName(),
+      null,
+      _ -> player.getPermissionLevel() > 0);
```
Comment (derklaro) [inline modules/bridge/src/main/java/eu/cloudnetservice/modules/bridge/platform/minestom/MinestomBridgeManagement.java]: This should probably use the permission function as well.

File: modules/bridge/src/main/java/eu/cloudnetservice/modules/bridge/platform/minestom/MinestomBridgeManagement.java [hunk 2]
```diff
@@ -141,7 +152,7 @@ public boolean isOnAnyFallbackInstance(@NonNull Player player) {
 
   @Override
   public @NonNull Optional<ServiceInfoSnapshot> fallback(@NonNull Player player, @Nullable String currServer) {
-    return this.fallback(player.getUuid(), currServer, null, player::hasPermission);
+    return this.fallback(player.getUuid(), currServer, null, _ -> player.getPermissionLevel() > 0);
```
Comment (derklaro) [inline modules/bridge/src/main/java/eu/cloudnetservice/modules/bridge/platform/minestom/MinestomBridgeManagement.java]: This should probably use the permission function as well.

File: modules/npcs/build.gradle.kts [hunk 3]
```diff
@@ -52,6 +52,11 @@ repositories {
       includeGroup("com.github.retrooper")
     }
   }
+  maven("https://repo.codemc.io/repository/maven-snapshots/") {
+    mavenContent {
+      includeGroup("com.github.retrooper")
+    }
+  }
```
Comment (derklaro) [inline modules/npcs/build.gradle.kts]: not needed anymore

**Review by derklaro — CHANGES_REQUESTED** [review]

**Review by derklaro — APPROVED** [review]

**Comment by DasBabyPixel** [comment]
Bei MinestomPlayerManagementListener handleLogin ist das PlayerSpawnEvent verwendet. Dadurch wird ein Spieler erst sehr spät im login wegen Wartungsarbeiten gekickt. Wäre AsyncPlayerConfigurationEvent nicht die bessere wahl?

**Pull request merged**